Bds b1c

andrewkamble88 opened the pull request:
:+1::tada: Hello, and thanks for contributing to [GNSS-SDR](http://gnss-sdr.org)! :tada::+1:

Before submitting your pull request, please make sure the following is done:
 1. You undertake the [Contributor Covenant Code of Conduct](https://github.com/gnss-sdr/gnss-sdr/blob/master/CODE_OF_CONDUCT.md).
 2. If you are a first-time contributor, after your pull request you will be asked to sign an Individual Contributor License Agreement ([CLA](https://en.wikipedia.org/wiki/Contributor_License_Agreement)) before your code gets accepted into `master`. This license is for your protection as a Contributor as well as for the protection of [CTTC](http://www.cttc.es/); it does not change your rights to use your own contributions for any other purpose. Except for the license granted therein to CTTC and recipients of software distributed by CTTC, you reserve all right, title, and interest in and to your contributions. The information you provide in that CLA will be maintained in accordance with [CTTC's privacy policy](http://www.cttc.es/privacy/).
 3. You have read the [Contributing Guidelines](https://github.com/gnss-sdr/gnss-sdr/blob/master/CONTRIBUTING.md).
 4. You have read the [coding style guide](http://gnss-sdr.org/coding-style/).
 5. You have forked the [gnss-sdr upstream repository](https://github.com/gnss-sdr/gnss-sdr) and have created your branch from `next` (or any other currently living branch in the upstream repository).
 6. Please include a description of your changes here.

**Please feel free to delete this line and the above text once you have read it and in case you want to go on with your pull request.**

andrewkamble88 committed:
Merge pull request #1 from gnss-sdr/next

Next

andrewkamble88 committed:
Merge pull request #2 from gnss-sdr/next

Next

andrewkamble88 committed:
Merge pull request #3 from andrewkamble88/next

Next

andrewkamble88 committed:
Add files via upload

andrewkamble88 committed:
Add files via upload

System Parameters for Beidou B1C Code generation.

andrewkamble88 commented:
Added System Parameters file: "Beidou_B1c.h"
Added Code generation files: "beidou_b1c_signal_processing.h" and "beidou_b1c_signal._processing.cc"

dmiralles2009 commented on src/algorithms/libs/beidou_b1c_signal_processing.cc:
```diff
@@ -0,0 +1,427 @@
+//=========================================SECONDARY CODE GENERATION OF PILOT COMPONENT=========================================
+
+//! Generate a version of the B1c Pilot Secondary code
+void make_b1cpSecondary(int32_t* _dest, int prn)
```
can you rename this to make_b1cp_secondary?. Better to keep consistency in the naming. Errors in B2a has been fixed

dmiralles2009 commented on src/algorithms/libs/beidou_b1c_signal_processing.cc:
```diff
@@ -0,0 +1,427 @@
+}
+
+//! Generate a complex version of the B1c Pilot Secondary code
+void beidou_b1cp_code_gen_complexSecondary(std::complex<float>* _dest, unsigned int _prn)
```
same issue with the naming

dmiralles2009 commented on src/core/system_parameters/Beidou_B1c.h:
```diff
@@ -0,0 +1,156 @@
+
+//!====================================PRIMARY CODE PARAMETERS OF B1C DATA COMPONENTS=============================================
+//!Phase difference(w) table data for Data Component
+const int32_t BEIDOU_B1cd_Phase_Diff[63] = {
```
Not a big issue but if you could keep constant names in capital letters will be great. Instead try naming this variable as 'BEIDOU_B1cd_PHASE_DIFF'

dmiralles2009 commented on src/core/system_parameters/Beidou_B1c.h:
```diff
@@ -0,0 +1,156 @@
+
+
+
+const double BEIDOU_B1cd_CODE_RATE_HZ = 10.23e6;      //!< BEIDOU_B1c data code rate [chips/s]
```
The official signal name is 'B1C', so if you could put the c in capital letters will be amazing. BeiDou B2a uses small 'a' and that is the reason why we used that way before

dmiralles2009 commented on src/core/system_parameters/Beidou_B1c.h:
```diff
@@ -0,0 +1,156 @@
+const int32_t BEIDOU_B1cp_SECONDARY_CODE_LENGTH_CHIPS = 1800;  // Each secondary code of Pilot component has the length 1800 chips.
+
+
+const int32_t WEIL_N = 10243;
```
Consider adding 'BEIDOU_B1C_' or similar in front of these variables in order to achieve consistency with variables within the file.

dmiralles2009 reviewed (commented):
Andrew, 
I am very happy with the progress so far. I suggested small fixes, but mostly cosmetic ones.

andrewkamble88 committed:
Update beidou_b1c_signal_processing.h

andrewkamble88 committed:
Update beidou_b1c_signal_processing.h

andrewkamble88 committed:
Update beidou_b1c_signal_processing.cc

andrewkamble88 committed:
Update beidou_b1c_signal_processing.cc

andrewkamble88 committed:
Update Beidou_B1c.h

andrewkamble88 committed:
Update beidou_b1c_signal_processing.cc

andrewkamble88 committed:
Update Beidou_B1c.h

andrewkamble88 committed:
Update beidou_b1c_signal_processing.cc

andrewkamble88 committed:
Add files via upload

andrewkamble88 committed:
Delete beidou_b1c_signal_processing.h

andrewkamble88 committed:
Add files via upload

andrewkamble88 committed:
Delete beidou_b1c_signal_processing.cc

andrewkamble88 committed:
Add files via upload

andrewkamble88 committed:
Delete Beidou_B1c.h

andrewkamble88 committed:
Add files via upload

andrewkamble88 committed:
Update beidou_b1c_signal_processing.cc

andrewkamble88 commented:
Renamed the variables at all appropriate positions in all files.
Thanks.

andrewkamble88 committed:
Update beidou_b1c_signal_processing.cc

andrewkamble88 committed:
Update Beidou_B1C.h

andrewkamble88 committed:
Update CMakeLists.txt

andrewkamble88 committed:
Update CMakeLists.txt

andrewkamble88 committed:
Add files via upload

beidou_b1cd_pcps_acquisition.h

andrewkamble88 committed:
Add files via upload

beidou_b1cd_pcps_acquisition.cc

andrewkamble88 committed:
Add files via upload

Added gnss-sdr_BDS_B1C_byte.conf file

andrewkamble88 committed:
Update gnss-sdr_BDS_B1C_byte.conf

andrewkamble88 committed:
Update beidou_b1cd_pcps_acquisition.cc

andrewkamble88 committed:
Update gnss-sdr_BDS_B1C_byte.conf

andrewkamble88 committed:
Update gnss-sdr_BDS_B1C_byte.conf

dmiralles2009 commented on conf/gnss-sdr_BDS_B1C_byte.conf:
```diff
@@ -0,0 +1,125 @@
+Channels.in_acquisition=1
+Channel.signal=C1
+
+Channel0.satellite = 6;
```
For testing I will remove this line, you do not know beforehand what satellites are available in the data set :)

andrewkamble88 replied:
Yes:) Removed all assigned satellites.

dmiralles2009 commented on conf/gnss-sdr_BDS_B1C_byte.conf:
```diff
@@ -0,0 +1,125 @@
+Channel9.satellite = 5;
+
+;######### ACQUISITION GLOBAL CONFIG ############
+Acquisition_C1.implementation=BEIDOU_B1C_PCPS_Acquisition
```
To test the blocks you could write unit test or test the code using the tracking and telemetry placeholdes

andrewkamble88 committed:
Update gnss-sdr_BDS_B1C_byte.conf

andrewkamble88 commented on src/algorithms/acquisition/adapters/beidou_b1cd_pcps_acquisition.cc:
```diff
@@ -0,0 +1,332 @@
+            item_size_ = sizeof(gr_complex);
+        }
+
+    acq_parameters_.ms_per_code = 1;
```
Hey Damian, Do we need to change this value to "10"? as code period in Beidou B1C is 10ms

andrewkamble88 committed:
Update gnss_block_factory.h

andrewkamble88 committed:
Update gnss_block_factory.cc

andrewkamble88 committed:
Update gnss_flowgraph.h

andrewkamble88 committed:
Update gnss_flowgraph.cc

andrewkamble88 committed:
Update dll_pll_veml_tracking.h

dmiralles2009 commented:
I like your code development so far. It looks good. Remember to update your bds_b1c branch with contents from the gnss-sdr next branch so that you work with the latest code always.

andrewkamble88 committed:
Add files via upload

andrewkamble88 committed:
Add files via upload

andrewkamble88 committed:
Merge pull request #5 from gnss-sdr/next

Next

andrewkamble88 committed:
Update dll_pll_veml_tracking.cc

andrewkamble88 committed:
Add files via upload

Added tracking adapter for B1C

andrewkamble88 committed:
Updated gnss_block_factory.cc for tracking of Beidou B1C

andrewkamble88 committed:
Add files via upload

dmiralles2009 commented on src/core/system_parameters/Beidou_B1C.h:
```diff
@@ -34,16 +34,32 @@
+const double BEIDOU_B1C_FREQ_HZ = FREQ1;              //!< BEIDOU B1C carrier frequency [Hz]
+
+
+const double BEIDOU_B1Cd_CODE_RATE_HZ = 10.23e6;      //!< BEIDOU_B1C data code rate [chips/s]
```
Please check the datasheet because the code rate for B1C is 1.023e6 instead of 10.23e6.

andrewkamble88 replied:
You are right! Updated to correct value.

dmiralles2009 commented on src/core/system_parameters/Beidou_B1C.h:
```diff
@@ -34,16 +34,32 @@
+
+
+const double BEIDOU_B1Cd_CODE_RATE_HZ = 10.23e6;      //!< BEIDOU_B1C data code rate [chips/s]
+const int32_t BEIDOU_B1Cd_CODE_LENGTH_CHIPS = 10230;  //!< BEIDOU_B1C data  code length [chips]
```
Same comment as before

andrewkamble88 replied:
Updated to correct value.

andrewkamble88 committed:
Updated with respect to Beidou B1C

andrewkamble88 committed:
Add files via upload

andrewkamble88 committed:
Used telemetry decoder of GPS L5

andrewkamble88 committed:
Update gnss-sdr_BDS_B1C_byte.conf

andrewkamble88 commented:
Hey Damian,

I have one doubt regarding the first evaluation. Do we need to submit any
code or documentation regarding what we have developed until first
evaluation to Google?

I received a reminder email from Google regarding the first evaluation. I
also overviewed questions they generally ask.

Please, can you guide me regarding how first evaluation occurs?



Regards,



Andrew Kamble


On Fri, Jun 21, 2019 at 11:22 PM Damian Miralles <notifications@github.com>
wrote:

> *@dmiralles2009* commented on this pull request.
> ------------------------------
>
> In src/core/system_parameters/Beidou_B1C.h
> <https://github.com/dmiralles2009/gnss-sdr/pull/4#discussion_r296334557>:
>
> >
> -const double BEIDOU_B1Cd_CODE_RATE_HZ = 10.23e6;      //!< BEIDOU_B1c data code rate [chips/s]
> -const int32_t BEIDOU_B1Cd_CODE_LENGTH_CHIPS = 10230;  //!< BEIDOU_B1c data  code length [chips]
> +
> +const double BEIDOU_B1C_FREQ_HZ = FREQ1;              //!< BEIDOU B1C carrier frequency [Hz]
> +
> +
> +const double BEIDOU_B1Cd_CODE_RATE_HZ = 10.23e6;      //!< BEIDOU_B1C data code rate [chips/s]
>
> Please check the datasheet because the code rate for B1C is 1.023e6
> instead of 10.23e6.
> ------------------------------
>
> In src/core/system_parameters/Beidou_B1C.h
> <https://github.com/dmiralles2009/gnss-sdr/pull/4#discussion_r296334615>:
>
> >
> -const double BEIDOU_B1Cd_CODE_RATE_HZ = 10.23e6;      //!< BEIDOU_B1c data code rate [chips/s]
> -const int32_t BEIDOU_B1Cd_CODE_LENGTH_CHIPS = 10230;  //!< BEIDOU_B1c data  code length [chips]
> +
> +const double BEIDOU_B1C_FREQ_HZ = FREQ1;              //!< BEIDOU B1C carrier frequency [Hz]
> +
> +
> +const double BEIDOU_B1Cd_CODE_RATE_HZ = 10.23e6;      //!< BEIDOU_B1C data code rate [chips/s]
> +const int32_t BEIDOU_B1Cd_CODE_LENGTH_CHIPS = 10230;  //!< BEIDOU_B1C data  code length [chips]
>
> Same comment as before
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/dmiralles2009/gnss-sdr/pull/4?email_source=notifications&email_token=AI4TV7OLYZDBT5GWXGOOXFLP3UIO5A5CNFSM4HS5ATVKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOB4J4LBY#pullrequestreview-252953991>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AI4TV7I5CM4ODV5O35GAPR3P3UIO5ANCNFSM4HS5ATVA>
> .
>

andrewkamble88 committed:
Update gnss-sdr_BDS_B1C_byte.conf

dmiralles2009 commented on src/algorithms/tracking/gnuradio_blocks/dll_pll_veml_tracking.cc:
```diff
@@ -320,6 +323,31 @@ dll_pll_veml_tracking::dll_pll_veml_tracking(const Dll_Pll_Conf &conf_) : gr::bl
+                    d_code_period = BEIDOU_B1Cd_PERIOD;
+                    d_code_chip_rate = BEIDOU_B1Cd_CODE_RATE_HZ;
+                    d_symbols_per_bit = BEIDOU_B1C_SAMPLES_PER_SYMBOL;
+                    d_correlation_length_ms = 1;
```
These settings are incorrect. You need 10 ms correlations to properly track B1C code. The other variable also needs adjustments depending on its nav bit length

dmiralles2009 committed:
Merge branch 'bds_b1c' of https://github.com/andrewkamble88/gnss-sdr into andrewkamble88-bds_b1c

dmiralles2009 committed:
bds_b1c: Fixes bugs with implementation

Fixes bugs with code for its implementation, adjust tracking values and follows
model of galileo ambiguous acquisition.

dmiralles2009 committed:
bds_b1c: quick fixes for data plotting

dmiralles2009 committed:
bds_b1c: fixed bug with pilot code generation

Pilot code sits with 90 degree phase and as a result it needs to be placed in
the imaginary component of the signal generation process. Code still needs to
account for modulation shceme of the signal with the sineBOC components

andrewkamble88 committed:
Merge pull request #6 from dmiralles2009/andrewkamble88-bds_b1c

Fixing issues with memory buffer, signals id, etc

andrewkamble88 committed:
Added function generating complex BEIDOU B1C data+pilot code

andrewkamble88 committed:
Added function generating BEIDOU B1C data+pilot code

andrewkamble88 committed:
Added function for "acq_iq_"

andrewkamble88 committed:
Update beidou_b1c_signal_processing.cc

andrewkamble88 committed:
Update beidou_b1c_signal_processing.cc

andrewkamble88 committed:
Update beidou_b1c_pcps_acquisition.cc

andrewkamble88 committed:
Merge pull request #7 from gnss-sdr/next

Next

andrewkamble88 committed:
Add files via upload

Updated with new gsl settings, added efficient algorithm for code generation, adding sine BOC to code

andrewkamble88 committed:
Update beidou_b1c_signal_processing_new.h

andrewkamble88 committed:
Update beidou_b1c_signal._processing_new.cc

andrewkamble88 committed:
Changing code with new gsl settings

andrewkamble88 committed:
Changed with new "gsl" settings

andrewkamble88 committed:
Updated with new "gsl" settings

dmiralles2009 commented on src/algorithms/acquisition/adapters/CMakeLists.txt:
```diff
@@ -36,6 +36,8 @@ set(ACQ_ADAPTER_SOURCES
     glonass_l2_ca_pcps_acquisition.cc
     beidou_b1i_pcps_acquisition.cc
     beidou_b3i_pcps_acquisition.cc
+    beidou_b1cd_pcps_acquisition.cc
```
This version of the file should probably be removed.

dmiralles2009 commented on src/algorithms/acquisition/adapters/CMakeLists.txt:
```diff
@@ -57,6 +59,8 @@ set(ACQ_ADAPTER_HEADERS
     glonass_l2_ca_pcps_acquisition.h
     beidou_b1i_pcps_acquisition.h
     beidou_b3i_pcps_acquisition.h
+    beidou_b1cd_pcps_acquisition.h
```
Same as above, this version should be removed

andrewkamble88 committed:
Updated with new "gsl" settings and modified algorithms to increase efficiency

andrewkamble88 committed:
Updated Code with CBOC functions

andrewkamble88 committed:
fixed some typos

andrewkamble88 committed:
Update beidou_b1c_signal_processing.cc

andrewkamble88 committed:
Added function for "Data+Pilot" CBOC code generation

andrewkamble88 committed:
Delete beidou_b1c_signal._processing_new.cc

andrewkamble88 committed:
Delete beidou_b1c_signal_processing_new.h

dmiralles2009 commented on src/algorithms/libs/beidou_b1c_signal_processing.cc:
```diff
@@ -0,0 +1,859 @@
+	// Generate Data Primary code
+	std::deque<bool> b1cd_primary_code =  make_b1cd_primary_weil_seq(phase_diff, truncation_point);
+
+	for (uint32_t n = 0; n < 10230; n++)
```
Avoid using hardcoded values like 10230. Use instead the constant defining the code length

dmiralles2009 commented on src/algorithms/libs/beidou_b1c_signal_processing.cc:
```diff
@@ -0,0 +1,859 @@
+{
+	const uint32_t _codeLength = 12 * BEIDOU_B1Cd_CODE_LENGTH_CHIPS;
+	
+	const float alpha = (1.0 / 2.0);
```
Add comment why alpha is 1/2, state eq in ICD

dmiralles2009 commented on src/algorithms/libs/beidou_b1c_signal_processing.cc:
```diff
@@ -0,0 +1,859 @@
+//! Generates float version of sine BOC(1,1) modulated Data Code 
+void beidou_b1cd_gen_float_11(gsl::span<float> _dest, gsl::span<int> _prn)
+{
+	const uint32_t _codeLength = 12 * BEIDOU_B1Cd_CODE_LENGTH_CHIPS;
```
Use a variable for this or add a comment to this value. Reading 12 is not clear where you got that

dmiralles2009 commented on src/algorithms/libs/beidou_b1c_signal_processing.cc:
```diff
@@ -0,0 +1,859 @@
+	
+	const float alpha = (1.0 / 2.0);
+	
+	int32_t sinboc_11[12 * 10230] = {0}; //  _codeLength not accepted by Clang
```
Avoid using hardcoded values, use variables with self-explanatory name

dmiralles2009 commented on src/algorithms/libs/beidou_b1c_signal_processing.cc:
```diff
@@ -0,0 +1,859 @@
+	
+	const float alpha = sqrt(1.0 / 11.0);
+	
+	int32_t sinboc_61[12 * 10230] = {0};    //  _codeLength not accepted by Clang
```
Same comment as before, avoid hard coded values

dmiralles2009 commented on src/algorithms/acquisition/adapters/beidou_b1c_pcps_acquisition.cc:
```diff
@@ -0,0 +1,387 @@
+    // Perform acquisition in Data + Pilot signal
+    if (acq_iq_)
+        {	
+	        //beidou_b1c_code_gen_complex_sampled(code, gnss_synchro_->PRN, fs_in_);
```
You should add the capability to perform acquisition with both codes

dmiralles2009 commented on src/algorithms/acquisition/adapters/beidou_b1c_pcps_acquisition.cc:
```diff
@@ -0,0 +1,387 @@
+    // Perform acquisition in Pilot signal
+    else if (acq_pilot_)
+        {
+    		beidou_b1cp_code_gen_complex_sampled(code, gnss_synchro_->PRN, fs_in_);
```
This function calls do not reflect the new GSL or BOC addition, consider fixing!

dmiralles2009 commented on src/algorithms/libs/beidou_b1c_signal_processing.cc:
```diff
@@ -0,0 +1,859 @@
+}
+
+
+void beidou_b1cp_code_gen_float_sampled_boc_11(gsl::span<float> _dest, uint32_t _prn)
```
The functions need to be added to .h file so that it can be accessed from external files. If you fix the acquisition comments you will notice this

dmiralles2009 commented on src/core/receiver/gnss_block_factory.cc:
```diff
@@ -39,6 +39,9 @@
 #include "array_signal_conditioner.h"
 #include "beamformer_filter.h"
+#include "beidou_b1c_dll_pll_tracking.h"
+#include "beidou_b1cd_pcps_acquisition.h"
```
Maybe this file should be removed, the beidou_b1c_pcps_acquisition takes care of the work now

dmiralles2009 reviewed (commented):
Hey Andrew, some cosmetic changes are needed. Good job :)

dmiralles2009 commented on src/algorithms/acquisition/adapters/beidou_b1c_pcps_acquisition.cc:
```diff
@@ -0,0 +1,387 @@
+}
+
+
+BeidouB1cPcpsAcquisition::~BeidouB1cPcpsAcquisition()
```
With gsl memory allocation, this changes. Check how B2a does it

dmiralles2009 commented on src/algorithms/acquisition/adapters/beidou_b1c_pcps_acquisition.cc:
```diff
@@ -0,0 +1,387 @@
+    auto* code = new std::complex<float>[code_length_];
+    // Perform acquisition in Data + Pilot signal
+    if (acq_iq_)
+        {	
```
Also, here you should be calling the boc functions

dmiralles2009 commented on src/algorithms/acquisition/adapters/beidou_b1cd_pcps_acquisition.cc:
```diff
@@ -0,0 +1,332 @@
+/*!
```
This file should be removed. The b1c_pcps_acq does the work

dmiralles2009 commented on src/algorithms/libs/beidou_b1c_signal_processing.cc:
```diff
@@ -0,0 +1,859 @@
+
+    beidou_b1cd_gen_float_11(_signal_B1C_data_span, gsl::span<int>(b1c_data_primary_code_chips, static_cast<uint32_t>(BEIDOU_B1Cd_CODE_LENGTH_CHIPS)));  // generate cboc 12 samples per chip
+    
+    volk_gnsssdr_free(b1c_data_primary_code_chips);
```
Avoid this memory allocation scheme, and keep same consistency as per the allocations you did before. See how B2a does this, where you define a code length and then allocate an array  based on that size

dmiralles2009 commented on src/algorithms/libs/beidou_b1c_signal_processing.cc:
```diff
@@ -0,0 +1,859 @@
+                    _dest[i] = std::complex<float>(1.0 - 2.0 * real_code_span[_codeValueIndex], 0);  //repeat the chip -> upsample
+                }
+        }           
+    volk_gnsssdr_free(real_code);
```
Same comment as before regarding this type of memory allocation. Keep the consistency you developed before

dmiralles2009 commented on src/algorithms/libs/beidou_b1c_signal_processing.cc:
```diff
@@ -0,0 +1,859 @@
+
+    beidou_b1cp_gen_float_11(_signal_B1C_pilot_span, gsl::span<int>(b1c_pilot_primary_code_chips, static_cast<uint32_t>(BEIDOU_B1Cp_CODE_LENGTH_CHIPS)));  // generate cboc 12 samples per chip
+    
+    volk_gnsssdr_free(b1c_pilot_primary_code_chips);
```
Same issue with dynamic memory allocation inside a function, keep consistency

dmiralles2009 commented on src/algorithms/libs/beidou_b1c_signal_processing.h:
```diff
@@ -0,0 +1,85 @@
+
+//! Generates complex BEIDOU B1C data+pilot code for the desired SV ID and sampled to specific sampling frequency
+void beidou_b1c_code_gen_complex_sampled(gsl::span<std::complex<float>> _dest, uint32_t _prn, int32_t _fs);
+
```
The version of the code with the BOC functions need to be added in .h file so that it can be accessed by external files

dmiralles2009 commented on src/algorithms/tracking/gnuradio_blocks/dll_pll_veml_tracking.cc:
```diff
@@ -708,6 +736,24 @@ void dll_pll_veml_tracking::start_tracking()
+    		if (trk_parameters.track_pilot)
+                {
+    			    // Secondary pilot code is specific for each satellite
+    			    d_secondary_code_string = const_cast<std::string *>(&BEIDOU_B1Cp_SECONDARY_CODE[d_acquisition_gnss_synchro->PRN - 1]);
```
BOC version of the code generation are needed here

dmiralles2009 commented on src/core/system_parameters/Beidou_B1C.h:
```diff
@@ -0,0 +1,242 @@
+const double  BEIDOU_B1Cp_PERIOD = 0.01;               //!< BEIDOU_B1C data code period [seconds]
+const int32_t BEIDOU_B1Cp_PERIOD_MS = 10;
+
+const int32_t BEIDOU_NBR_SATS = 63;		      //!<Total number of satellites
```
This label is very generic, consider renaming to something like BEIDOU_B1C_NBR_SATS to avoid issues when integrating the code

dmiralles2009 commented on src/core/system_parameters/Beidou_B1C.h:
```diff
@@ -0,0 +1,242 @@
+
+//!====================================PRIMARY CODE PARAMETERS OF B1C DATA COMPONENTS=============================================
+//!Phase difference(w) table data for Data Component
+const int32_t BEIDOU_B1Cd_PHASE_DIFF[63] = {
```
Use constant value hre BEIDOU_B1C_NBR_SATS

dmiralles2009 reviewed (commented):
Andrew, as stated in the email, this version of the code does not even compile. Please fix all issues mentioned and make sure compilation happens for this code. That is the only way I could help debug things out

andrewkamble88 committed:
Changed variable d_correlation_length_ms accoording to B1C settings

andrewkamble88 committed:
Removed old acquisition adapter files for B1C

andrewkamble88 committed:
Given comments and equation in BOC code for more understanding and removed hard coded values

andrewkamble88 committed:
Changed the constant "BEIDOU_NBR_SATS" to "BEIDOU_B1C_NBR_SATS"

andrewkamble88 committed:
Delete beidou_b1cd_pcps_acquisition.cc

andrewkamble88 committed:
Delete beidou_b1cd_pcps_acquisition.h

andrewkamble88 committed:
Added BOC functions

andrewkamble88 committed:
Added new BOC functions and changed according to new "gsl" settings

andrewkamble88 committed:
Allocated memory like b2a

andrewkamble88 committed:
Update beidou_b1c_signal_processing.cc

andrewkamble88 committed:
Update beidou_b1c_signal_processing.cc

andrewkamble88 committed:
Update beidou_b1c_signal_processing.cc

andrewkamble88 committed:
Added function "beidou_b1c_code_gen_sinboc11_float" for pilot code

andrewkamble88 committed:
Added function "beidou_b1c_code_gen_sinboc11_float" for tracking

andrewkamble88 committed:
Added function "beidou_b1c_code_gen_sinboc11_float" for tracking

andrewkamble88 committed:
Removed references to deleted files

andrewkamble88 committed:
Changed type of "code_"

andrewkamble88 committed:
Update beidou_b1c_signal_processing.cc

andrewkamble88 committed:
Update beidou_b1c_pcps_acquisition.cc

andrewkamble88 committed:
Update beidou_b1c_signal_processing.cc

andrewkamble88 committed:
Update beidou_b1c_signal_processing.h

andrewkamble88 committed:
Update dll_pll_veml_tracking.cc

andrewkamble88 committed:
Update beidou_b1c_signal_processing.cc

andrewkamble88 committed:
Update beidou_b1c_signal_processing.h

andrewkamble88 committed:
Removed "float_sampled_boc" function

andrewkamble88 committed:
Rempved "float_sampled_boc" function

andrewkamble88 committed:
updated with "beidou_b1c_signal_processing.cc"

dmiralles2009 commented on src/algorithms/acquisition/adapters/beidou_b1c_pcps_acquisition.cc:
```diff
@@ -255,12 +255,12 @@ void BeidouB1cPcpsAcquisition::set_local_code()
     // Perform acquisition in Pilot signal
```
You need to enable acq_iq_. I think the error you mentioned was fixed in our emails by applying the ICD equation when combining and doing addition

andrewkamble88 replied:
Yes, I enabled "acq_iq_".

dmiralles2009 commented on src/algorithms/libs/beidou_b1c_signal_processing.cc:
```diff
@@ -0,0 +1,914 @@
+        {
+            for (uint32_t n = 0; n < 12 * BEIDOU_B1Cd_CODE_LENGTH_CHIPS; n++)
+                {
+                    // XORing gives error as two values generating are fractional
```
Have you try doing the addition here?. This is what we agreed upon in our email exchange

andrewkamble88 replied:
Yes, by doing addition, the code is compiling properly.

andrewkamble88 committed:
Removed function "float_sampled_boc"

andrewkamble88 committed:
Inserted "addition" instead of "XOR" in combining "data+ pilot" code

andrewkamble88 committed:
Enabled "acq_iq_"

andrewkamble88 committed:
Removed "float_sampled" function

andrewkamble88 committed:
Remobed functions "float_sampled_boc"

dmiralles2009 commented on src/algorithms/libs/beidou_b1c_signal_processing.cc:
```diff
@@ -0,0 +1,913 @@
+            if (i == _samplesPerCode - 1)
+                {
+                    //--- Correct the last index (due to number rounding issues) -----------
+                    _dest[i] = std::complex<float>(1.0 - 2.0 * real_code_span[_codeLength - 1], 1.0 - 2.0 * imaginary_code_span[_codeLength - 1]);
```
You are using the index _codeLength here, which translates to always using the same index. This seems to be a typo, consider fixing

dmiralles2009 commented on src/algorithms/libs/beidou_b1c_signal_processing.cc:
```diff
@@ -0,0 +1,913 @@
+                }
+            else
+                {
+                    _dest[i] = std::complex<float>(1.0 - 2.0 * real_code_span[_codeValueIndex], 1.0 - 2.0 * imaginary_code_span[_codeLength - 1]);  //repeat the chip -> upsample
```
Same comment as before, for one case you use `_codeValueIndex`, but for the other you use `_codeLength`, this does not seem right

dmiralles2009 commented on src/algorithms/libs/beidou_b1c_signal_processing.cc:
```diff
@@ -0,0 +1,913 @@
+            if (i == _samplesPerCode - 1)
+                {
+                    //--- Correct the last index (due to number rounding issues) -----------
+                    _dest[i] = std::complex<float>(1.0 - 2.0 * real_code_span[_codeLength - 1], 1.0 - 2.0 * imaginary_code_span_pilot[_codeLength - 1]);
```
Same comments as before, is this an error?

dmiralles2009 commented on src/algorithms/libs/beidou_b1c_signal_processing.cc:
```diff
@@ -0,0 +1,913 @@
+    volk_gnsssdr_free(real_code_data);
+    volk_gnsssdr_free(real_code_pilot);
+    volk_gnsssdr_free(imaginary_code_pilot);
+    volk_gnsssdr_free(real_code);
```
The name seems misleading, consider using something like `combined_real_code` or similar

dmiralles2009 commented on src/algorithms/libs/beidou_b1c_signal_processing.cc:
```diff
@@ -0,0 +1,913 @@
+}
+
+
+//! Generates Data code required at the time of tracking(followed approach like Galileo E1)
```
Code comments need to be fixed

dmiralles2009 commented on src/algorithms/libs/beidou_b1c_signal_processing.cc:
```diff
@@ -0,0 +1,913 @@
+}
+
+
+//! Generates pilot code required at the time of tracking(followed approach like Galileo E1)
```
Same issue with code comments

dmiralles2009 reviewed (commented):
Andrew,
multiple errors in this revision. You need to be more careful managing buffers and indexes, copy and paste might be dangerous on occasions.

You will also need to start testing your code more efficiently, some errors are just too basic for you to disregard.

Consider also fixing all comments created before as some of them are still pending

andrewkamble88 committed:
Fixed the typos

dmiralles2009 commented on src/algorithms/libs/beidou_b1c_signal_processing.cc:
```diff
@@ -873,7 +873,7 @@ void beidou_b1c_code_gen_complex_sampled_boc(gsl::span<std::complex<float>> _des
             else
                 {
-                    _dest[i] = std::complex<float>(1.0 - 2.0 * real_code_span[_codeValueIndex], 1.0 - 2.0 * imaginary_code_span_pilot[_codeLength - 1]);  //repeat the chip -> upsample
+                    _dest[i] = std::complex<float>(1.0 - 2.0 * real_code_span[_codeValueIndex], 1.0 - 2.0 * imaginary_code_span_pilot[_codeValueIndex - 1]);  //repeat the chip -> upsample
```
Still, inconsistencies shown here, why `[_codeValueIndex - 1]` in the pilot but not in the data component?

dmiralles2009 reviewed (commented):
Slow down when developing code. This kind of practices is what causes memory errors in the code

andrewkamble88 committed:
Fixed the typos

andrewkamble88 committed:
Added fields of the navigation message B-CNAV1 of Beidou B1C

andrewkamble88 committed:
Added additional subframe  constants

andrewkamble88 committed:
Added system parameters files for CNAV1 navigation message

andrewkamble88 committed:
Added all the system parameters files for CNAV1 navigation message

dmiralles2009 commented on src/core/system_parameters/beidou_cnav1_navigation_message.cc:
```diff
@@ -0,0 +1,605 @@
+    	cnav1_ephemeris.IODE = static_cast<double>(read_navigation_unsigned(subframe2, IODE_S_2));
+    	
+        // Ephemeris I Start
+        cnav1_ephemeris.t_oe = static_cast<double>(read_navigation_unsigned(subframe2, t_oe_S_2)) * 300;                              //[s] effective range 0~604500
```
Make this a constant rather than a hardcoded number

andrewkamble88 replied:
Replaced hardcoded number by constant.

dmiralles2009 commented on src/core/system_parameters/beidou_cnav1_navigation_message.cc:
```diff
@@ -0,0 +1,605 @@
+    	//===============================================Decode string message of Subframe 1==============================================
+    	
+    	cnav1_ephemeris.i_satellite_PRN = static_cast<uint32_t>(read_navigation_unsigned(subframe1, PRN));
+    	cnav1_ephemeris.SOH = static_cast<uint32_t>(read_navigation_unsigned(subframe1, SOH)) * 18;   ///Scalling factor= 18, Unit= [s]
```
Make the 18 a constant rather than a hardcoded number

andrewkamble88 replied:
Replaced hardcoded number by constant.

dmiralles2009 commented on src/core/system_parameters/beidou_cnav1_navigation_message.cc:
```diff
@@ -0,0 +1,605 @@
+                    flag_all_ephemeris = true;
+                    // Update the time of ephemeris information
+                    d_previous_tb = cnav1_ephemeris.IODE;
+                    DLOG(INFO) << "Beidou Cnav1 Ephemeris from Subframe 2 have been received and belong to the same batch" << std::endl;
```
for printing make it BeiDou rather than Beidou

andrewkamble88 replied:
Made appropriate changes.

dmiralles2009 reviewed (commented):
Good, job, minimal cosmetic changes are needed

andrewkamble88 committed:
Done some cosmetic changes

andrewkamble88 committed:
Added two separate functions for CRC of Subframe 2 and Subframe 3

dmiralles2009 reviewed (commented):
So far so good. No major issues detected

andrewkamble88 committed:
Updated all system_parameters files related to telemetry_decoding

andrewkamble88 committed:
Added Adapter for telemetry_decoder of BeiDou B1C CNAV1 message and updated CMakeLists.txt

andrewkamble88 committed:
Added files within gnu_radio of telemetry_decoder of Beidou B1C CNAV1 navigation message and updated CMakeLists.txt

andrewkamble88 committed:
Connected telemetry_decoder of Beidou B1C with the signal flowgraph

andrewkamble88 committed:
Cleaning for final submission

andrewkamble88 committed:
Code cleaning for final submission

andrewkamble88 committed:
Added Matlab Script which used for generating Secondary Pilot Code for Tracking

andrewkamble88 committed:
Code Cleaning for GSoC final submission
